presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -394,7 +395,22 @@ public static boolean isS3FileSystem(HdfsContext context, HdfsEnvironment hdfsEn
     public static boolean isViewFileSystem(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
     {
         try {
-            return getRawFileSystem(hdfsEnvironment.getFileSystem(context, path)) instanceof ViewFileSystem;
+            FileSystem rawFileSystem = getRawFileSystem(hdfsEnvironment.getFileSystem(context, path));
```

Comment:
is it possible to put the logic in getRawFileSystem?   something like below. 
```
    private static FileSystem getRawFileSystem(FileSystem fileSystem)
    {
        if (fileSystem instanceof HadoopExtendedFileSystem) {
            return getRawFileSystem(((HadoopExtendedFileSystem) fileSystem).getRawFileSystem());
        } else if (fileSystem instanceof CachingFilesystem) {
            return getRawFileSystem(cachingfilesystem.getDatatier())
        }
        return fileSystem;
    }
```

Reply:
Ok, I think this has been implemented in https://github.com/prestodb/presto/pull/17390/files